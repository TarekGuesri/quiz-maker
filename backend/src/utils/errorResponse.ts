export class ErrorResponse extends Error {
  statusCode = 0;
  constructor(message: string, statusCode: number) {
    super(message);
    this.statusCode = statusCode;

    Error.captureStackTrace(this, this.constructor);
  }
}
