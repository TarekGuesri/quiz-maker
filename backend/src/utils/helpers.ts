export const trimmer = (text: string): string => {
  if (text) {
    return text.trim();
  } else {
    return text;
  }
};
