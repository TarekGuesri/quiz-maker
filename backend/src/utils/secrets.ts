import dotenv from "dotenv";
import fs from "fs";

if (fs.existsSync(".env")) {
  console.log("Using .env file to supply config environment variables");
  dotenv.config({ path: ".env" });
} else {
  console.log("Using .env file to supply config environment variables");
  dotenv.config({ path: ".env.example" }); // you can delete this after you create your own .env file!
}

export const ENVIRONMENT = process.env.NODE_ENV;
const prod = ENVIRONMENT === "production"; // Anything else is treated as 'dev'

export const MONGODB_URI = prod
  ? process.env["MONGODB_URI"]
  : process.env["MONGODB_URI_LOCAL"];

if (!MONGODB_URI) {
  if (prod) {
    console.error(
      "No mongo connection string. Set MONGODB_URI environment variable.",
    );
  } else {
    console.error(
      "No mongo connection string. Set MONGODB_URI_LOCAL environment variable.",
    );
  }
  process.exit(1);
}
