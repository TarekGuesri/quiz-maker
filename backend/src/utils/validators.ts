import { Schema } from "express-validator";

export const quizSchema: Schema = {
  title: {
    // isLength: { errorMessage: "Title can't be empty", options: { min: 1 } },
    errorMessage: "Title can't be empty!",
    notEmpty: true,
  },
  questions: {
    isArray: { options: { min: 1, max: 10 } },
    errorMessage: "Questions must be an array of a length between 1 and 10!",
  },
  "questions.*.content": {
    errorMessage: "Question content can't be empty!",
    notEmpty: true,
  },
};
