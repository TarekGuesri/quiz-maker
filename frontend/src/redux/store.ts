import { Action, ThunkAction, configureStore } from "@reduxjs/toolkit";

import ui from "./ui/ui-slice";

export const store = configureStore({
  reducer: {
    ui,
  },
  devTools: true,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
