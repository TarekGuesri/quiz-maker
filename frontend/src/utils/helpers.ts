export const timeFormatter = (
  seconds: number,
  minutes: number,
  hours: number,
): string => {
  const secondsString = seconds.toString().padStart(2, "0");
  const minutesString = minutes.toString().padStart(2, "0");

  if (hours === 0) {
    return `${minutesString}:${secondsString}`;
  } else {
    const hoursString = hours.toString().padStart(2, "0");

    return `${hoursString}:${minutesString}:${secondsString}`;
  }
};
