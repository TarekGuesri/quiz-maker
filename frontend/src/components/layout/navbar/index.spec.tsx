import React from "react";
import { cleanup, render, screen, fireEvent } from "@testing-library/react";
import { Router } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "src/redux/store";
import { history } from "src/utils/history";
import { Navbar } from ".";

describe("src/components/layout/navbar", () => {
  afterEach(() => cleanup());

  test("Render Navbar component", () => {
    const renderedLoading = render(
      <Provider store={store}>
        <Router history={history}>
          <Navbar />
        </Router>
      </Provider>,
    );

    expect(renderedLoading.container).toMatchSnapshot();
  });

  test("Check if dark mode changes after clicking the switch", () => {
    render(
      <Provider store={store}>
        <Router history={history}>
          <Navbar />
        </Router>
      </Provider>,
    );

    const {
      ui: { darkMode: currentDarkMode },
    } = store.getState();

    const darkModeSwitch = screen.getByRole("checkbox", { name: /ðŸŒ™/i });

    // Clicking on the switch button
    fireEvent.click(darkModeSwitch);

    expect(store.getState().ui.darkMode).toEqual(!currentDarkMode);
  });
});
