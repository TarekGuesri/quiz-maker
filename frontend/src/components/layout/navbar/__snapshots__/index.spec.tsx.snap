// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/layout/navbar Render Navbar component 1`] = `
<div>
  <div
    class="MuiBox-root css-1rcbgiz"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-pivuyh-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular makeStyles-toolbar-1 css-1t29gy6-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h4 css-756fpk-MuiTypography-root"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
            style="text-decoration: none;"
          >
            QuizMaker
          </a>
        </div>
        <div
          class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
            >
              <span
                class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
              >
                <input
                  class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                  type="checkbox"
                />
                <span
                  class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              ðŸŒ™
            </span>
          </label>
        </div>
      </div>
    </header>
  </div>
</div>
`;
